src/main.ts

```diff
@@ -439,23 +439,18 @@ ipcMain.handle(
   }
 )
 
-ipcMain.handle('har:open', async (_, fileName: string): Promise<HarFile> => {
-  console.info('har:open event received')
-  let fileHandle: FileHandle | undefined
-  try {
-    fileHandle = await open(path.join(RECORDINGS_PATH, fileName), 'r')
-    const data = await fileHandle?.readFile({ encoding: 'utf-8' })
-    // TODO: https://github.com/grafana/k6-studio/issues/277
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const har = await JSON.parse(data)
-
+ipcMain.handle(
+  'har:open',
+  async (_, fileName: string): Promise<HarWithOptionalResponse> => {
+    const data = await readFile(path.join(RECORDINGS_PATH, fileName), {
+      encoding: 'utf-8',
+      flag: 'r',
+    })
     // TODO: https://github.com/grafana/k6-studio/issues/277
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    return { name: fileName, content: har }
-  } finally {
-    await fileHandle?.close()
+    return JSON.parse(data)
```

Comment:
This makes me think about the scenario where the file is too big and electron might just drop the data when trying to send it, silently 😕

Reply:
I replied to the wrong thing 🤦 
yeah will be worth to explore streaming data in the future as we move more things 👍